pom.xml

```diff
@@ -1486,6 +1487,16 @@
       </activation>
     </profile>
 
+    <profile>
+      <id>isSpark3.0</id>
+<!--      for spark 3.0.x we need override the follow propeprties to package and run test-->
+      <properties>
+        <spark3.version>3.0.0</spark3.version>
+        <spark.version>${spark3.version}</spark.version>
+        <scalatest.version>3.0.1</scalatest.version>
+      </properties>
+    </profile>
```

Comment:
instead of a boolean flag, better be explicit about spark version and have a maven profile match to it. we can call it `-Dspark3.0` and keep `-Dspark3` points to the latest Spark 3 release.